test/test_main.py

```diff
@@ -11,7 +12,10 @@
     sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))
     from main import run
 
-test_files = glob.glob(os.path.join(os.path.dirname(__file__), "files/**/*.ifc"), recursive=True)
+test_for_rules = any(re.match(r"^[a-zA-Z]{3}\d{3}$", arg) for arg in sys.argv[1:])
+test_files = [fn for fn in glob.glob(os.path.join(os.path.dirname(__file__), 
```

Comment:
I still find it a bit confusing:

- `test_for_rules`: why is this a boolean, why not do `filter()` instead of `any()` to filter out all non-rule code argv and use them below
- the regex is lower or upper case but by doing `x in fn.lower()` actually an uppercase argument will not work
- some of the rule codes such as `alb` will arbitrarily match word fragments such as `albino` which will be rather confusing. Test files are in separate directories so why not filter only on that directory name instead of entire file path?

Reply:
It's modified in https://github.com/buildingSMART/ifc-gherkin-rules/pull/51/commits/377f4e65b7fc0fb5f023b74dc4e07c37180a327f

> why is this a boolean, why not do filter() instead of any() to filter out all non-rule code argv and use them below

Done.

> some of the rule codes such as alb will arbitrarily match word fragments such as albino which will be rather confusing. Test files are in separate directories so why not filter only on that directory name instead of entire file path?

Done. 

> the regex is lower or upper case but by doing x in fn.lower() actually an uppercase argument will not work

The aim was to also allow for uppercase input, such as `'Alb001'` or` 'ALB001' `instead of `'alb001'` It should be good now: the regex match will still allow for this uppercase and it later converts into lowercases to match the folder name (which are always in lowercases, such as in 'alb001'). 

Furthermore, there is a warning added in the case there is someone wants to test a non-existing rule code (e.g. `'alb005'`). This could be either a typo or it could mean that there are no test files added yet (or something else even). Nothing too spectecular, but it might help someone writing rules and testing it locally.